File: projects/native-apps/ios/App/LastViewedReadlist.swift
import Foundation

struct LastViewedReadlist {
	private let defaults: UserDefaults

	private enum Key {
		static let readlistHref = "readingList.lastViewedReadlistHref"
	}

	init(defaults: UserDefaults) {
		self.defaults = defaults
	}

	var href: String? {
		defaults.string(forKey: Key.readlistHref)
	}

	func remember(href: String) {
		defaults.set(href, forKey: Key.readlistHref)
	}

	func forget() {
		defaults.removeObject(forKey: Key.readlistHref)
	}
}

File: projects/native-apps/android/app/src/main/kotlin/com/readplace/android/core/LastViewedReadlist.kt
package com.readplace.android.core

/**
 * The last readlist the server confirmed as current, remembered so a cold launch
 * reopens it directly instead of flashing the entry point first. Written only when
 * a server response marks a readlist current — never on a menu tap the server has
 * not yet confirmed — and forgotten on deliberate sign-out (not on a forced or
 * expiry logout, which preserves it).
 */
class LastViewedReadlist(private val storage: ReaderChoiceStorage) {
	val href: String? get() = storage.getString(KEY)

	fun remember(href: String) {
		storage.putString(KEY, href)
	}

	fun forget() {
		storage.remove(KEY)
	}

	private companion object {
		const val KEY = "readingList.lastViewedReadlistHref"
	}
}
